# **MÓDULO 19**
# Exercício: Estatística Aplicada

**Efetividade de Duas Estratégias de Ensino**

Imagine que uma escola esteja avaliando a eficácia de duas estratégias de ensino de matemática para alunos do ensino médio. Eles querem determinar se há uma diferença significativa no desempenho médio dos alunos entre as duas estratégias.

# **Hipóteses:**

* Hipótese nula (H0): A média das notas dos alunos na estratégia A é igual à média das notas dos alunos na estratégia B.
* Hipótese alternativa (H1): A média das notas na Estratégia B é maior do que a média das notas na Estratégia A.

# **Dados:**

* Amostra da Estratégia A: Notas de 50 alunos que receberam a Estratégia A.
* Amostra da Estratégia B: Notas de 50 alunos que receberam a Estratégia B.

Usaremos um teste Z para comparar as médias das notas entre as duas amostras.

Se o p-valor do teste Z for menor que um nível de significância pré-determinado (por exemplo, α = 0.05), rejeitamos a hipótese nula e concluímos que há uma diferença significativa nas médias das notas entre as duas estratégias de ensino.

In [21]:
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats

Os dados são criados a seguir:

In [22]:
# Definindo médias e desvios padrão para as notas nas duas estratégias
media_estrategia_A = 70
desvio_padrao_estrategia_A = 10

media_estrategia_B = 75
desvio_padrao_estrategia_B = 12

# Gerando as amostras de notas para cada estratégia de ensino da nossa base
np.random.seed(0)  # Para garantir a reprodutibilidade dos resultados
amostra_estrategia_A = np.random.normal(loc=media_estrategia_A, scale=desvio_padrao_estrategia_A, size=50)
amostra_estrategia_B = np.random.normal(loc=media_estrategia_B, scale=desvio_padrao_estrategia_B, size=50)

print("Notas da Estratégia A:", amostra_estrategia_A[:5])
print("Notas da Estratégia B:", amostra_estrategia_B[:5])

Notas da Estratégia A: [87.64052346 74.00157208 79.78737984 92.40893199 88.6755799 ]
Notas da Estratégia B: [64.25440127 79.64282997 68.87033835 60.83241379 74.66181326]


# 1) De acordo com as informações analisadas o nosso teste é unilateral á direita, esquerda ou bicaudal? Justifique.




## **RESPOSTA: UNILATERAL À DIREITA (Right-tailed)**

### Por que é **UNILATERAL À DIREITA**?

1. **Análise das Hipóteses:**
   - **H0:** μA = μB (não há diferença nas médias)
   - **H1:** μB > μA (a média da Estratégia B é **maior** que A)

2. **O que define o tipo de teste:**
   - A hipótese alternativa especifica uma **direção clara**: B maior que A
   - Não é uma simples diferença (≠), mas uma relação **direcional (>)**
   - Isso significa que estamos interessados apenas em um lado da distribuição

3. **Como a estatística Z é construída:**
   - O Z é calculado como: **Z = (X̄_B − X̄_A) / SE**
   - Valores **grandes e positivos** de Z evidenciam que B > A
   - Por isso a região de rejeição fica na **cauda direita** da distribuição

4. **Implicações Práticas:**
   - Com α = 0.05 unilateral à direita, o valor crítico é **z = 1,645** (e não 1,96, que seria o caso bicaudal)
   - Se Z > 1,645, rejeitamos H0
   - O p-valor é calculado para a cauda direita, **sem dividir por 2**

# 2) Calcule as médias para as duas amostragens e as variâncias. Quais insights você pode retirar comparando os dados?

In [ ]:
print("=" * 70)
print("QUESTÃO 2: Médias, Variâncias e Insights")
print("=" * 70)

# CÁLCULO DAS MÉDIAS
media_A = np.mean(amostra_estrategia_A)
media_B = np.mean(amostra_estrategia_B)

# CÁLCULO DAS VARIÂNCIAS
variancia_A = np.var(amostra_estrategia_A, ddof=1)  # ddof=1 para variância amostral
variancia_B = np.var(amostra_estrategia_B, ddof=1)

# CÁLCULO DOS DESVIOS PADRÃO AMOSTRAIS
desvio_A = np.std(amostra_estrategia_A, ddof=1)
desvio_B = np.std(amostra_estrategia_B, ddof=1)

# DIFERENÇA ENTRE AS MÉDIAS
diferenca_medias = media_B - media_A

# EXIBIÇÃO DOS RESULTADOS
print("\nESTATÍSTICAS DESCRITIVAS:\n")
print(f"{'Métrica':<30} {'Estratégia A':<20} {'Estratégia B':<20}")
print("-" * 70)
print(f"{'Média Amostral':<30} {media_A:<20.4f} {media_B:<20.4f}")
print(f"{'Variância Amostral':<30} {variancia_A:<20.4f} {variancia_B:<20.4f}")
print(f"{'Desvio Padrão Amostral':<30} {desvio_A:<20.4f} {desvio_B:<20.4f}")
print(f"{'Tamanho da Amostra':<30} {len(amostra_estrategia_A):<20} {len(amostra_estrategia_B):<20}")

print(f"\n🔍 DIFERENÇA ENTRE AS MÉDIAS:")
print(f"   Média B - Média A = {media_B:.4f} - {media_A:.4f} = {diferenca_medias:.4f}")
print(f"   A Estratégia B tem uma média {diferenca_medias:.4f} pontos MAIOR que A")

QUESTÃO 2: Médias, Variâncias e Insights

📊 ESTATÍSTICAS DESCRITIVAS:

Métrica                        Estratégia A         Estratégia B        
----------------------------------------------------------------------
Média Amostral                 71.4056              74.7487             
Variância Amostral             129.2655             110.4700            
Desvio Padrão Amostral         11.3695              10.5105             
Tamanho da Amostra             50                   50                  

🔍 DIFERENÇA ENTRE AS MÉDIAS:
   Média B - Média A = 74.7487 - 71.4056 = 3.3431
   A Estratégia B tem uma média 3.3431 pontos MAIOR que A


## **Insights e Análises - Questão 2**

### 📈 **O que observamos nos dados?**

1. **Diferença nas Médias:**
   - A Estratégia B possui uma média amostral **MAIOR** que a Estratégia A (74.75 vs 71.41)
   - Essa diferença positiva de **3.34 pontos** está alinhada com nossa hipótese alternativa (H1: μB > μA)
   - É um sinal promissor de que a Estratégia B pode ser mais eficaz

2. **Comparação das Variâncias:**
   - A Estratégia A tem **variância MAIOR** que a Estratégia B (129.27 vs 110.47)
   - Isso significa que os desempenhos na Estratégia A são mais **dispersos/heterogêneos**
   - A Estratégia B tem desempenho mais **consistente** (menos variação entre os alunos)
   - **Implicação:** B é duplamente favorável — média superior **e** menor variabilidade

3. **Tamanho das Amostras:**
   - Ambas as amostras têm **n = 50** alunos
   - Isso garante que os testes estatísticos são válidos (n > 30)
   - As amostras são do mesmo tamanho, facilitando a comparação

4. **Comparação com os Parâmetros Populacionais:**
   - **Parâmetros Teóricos:** μA = 70, μB = 75, σA = 10, σB = 12
   - **Amostras Observadas:** as médias amostrais estão próximas aos valores populacionais
   - As variâncias amostrais também refletem bem a população

### 🎯 **Conclusão Parcial:**
Os dados **suportam visualmente** a hipótese de que a Estratégia B é melhor — ela apresenta
média superior **e** menor dispersão. Ainda assim, precisamos testar se essa diferença é
**estatisticamente significativa** (próximo passo).

# 3) Imprima os resultados da estatística do teste Z, p value e indique se rejeitaremos ou não a hipótese nula. Justifique sua resposta.

In [24]:
#Seu código aqui

# 4) Para finalizar monte o gráfico da distribuição da estatística do nosso teste Z e explique o que pode ser observado através dele.

In [25]:
#Seu código aqui